packages/amplify-category-function/src/provider-utils/awscloudformation/utils/packageLayer.ts

```diff
@@ -42,7 +52,7 @@ export async function packageLayer(context, resource) {
     output.on('close', () => {
       // check zip size is less than 250MB
       if (validFilesize(destination)) {
-        const zipName = `${resource.resourceName}-build.zip`;
+        const zipName = `${layerName}-build.zip`;
```

Comment:
I know it's not part of this PR, but I think I'd move all of this zipping logic to a helper function just to keep this function easily readable